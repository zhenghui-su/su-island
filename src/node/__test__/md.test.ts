import { unified } from 'unified';
import remarkParse from 'remark-parse';
import remarkRehype from 'remark-rehype';
import rehypeStringify from 'rehype-stringify';
import { describe, test, expect } from 'vitest';

// markdown语法编译测试
describe('Markdown complie cases', async () => {
  const processor = unified();

  processor.use(remarkParse).use(remarkRehype).use(rehypeStringify);
  // 测试编译标题
  test('Compile title', async () => {
    const mdTitle = '# 123';
    const result = processor.processSync(mdTitle);
    expect(result.value).toMatchInlineSnapshot('"<h1>123</h1>"');
  });
  // 测试编译代码块
  test('Compile code', async () => {
    const mdCode = 'I am using `su-island.js`';
    const result = processor.processSync(mdCode);
    expect(result.value).toMatchInlineSnapshot(
      '"<p>I am using <code>su-island.js</code></p>"'
    );
  });
});
