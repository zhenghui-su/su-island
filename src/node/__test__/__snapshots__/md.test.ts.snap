// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Markdown complie cases > Compile title 1`] = `"<h1>123</h1>"`;
